tripal/includes/TripalImporter.inc

```diff
@@ -0,0 +1,775 @@
+      throw new Exception('Cannot prepare the importer: ' . $e->getMessage());
+    }
+  }
+
+  /**
+   * Cleans up any temporary files that were created by the prepareFile().
+   *
+   * This function should be called after a run() to remove any temporary
+   * files and keep them from building up on the server.
+   */
+  public function cleanFile() {
+    try {
+      // If a remote file was downloaded then remove it.
+      for ($i = 0; $i < count($this->arguments['files']); $i++) {
+        if (!empty($this->arguments['files'][$i]['file_remote']) and
+          file_exists($this->arguments['files'][$i]['file_path'])) {
+          $this->logMessage('Removing downloaded file...');
+          unlink($this->arguments['files'][$i]['file_path']);
+          $this->is_prepared = FALSE;
+        }
+      }
+    } catch (Exception $e) {
+      throw new Exception('Cannot prepare the importer: ' . $e->getMessage());
+    }
+  }
+
+  /**
+   * Logs a message for the importer.
+   *
+   * There is no distinction between status messages and error logs.  Any
+   * message that is intended for the user to review the status of the loading
+   * can be provided here.  If this importer is associated with a job then
+   * the logging is passed on to the job for storage.
+   *
+   * Messages that are are of severity TRIPAL_CRITICAL or TRIPAL_ERROR
+   * are also logged to the watchdog.
+   *
+   * @param $message
+   *   The message to store in the log. Keep $message translatable by not
+   *   concatenating dynamic values into it! Variables in the message should
+   *   be added by using placeholder strings alongside the variables argument
+   *   to declare the value of the placeholders. See t() for documentation on
+   *   how $message and $variables interact.
+   * @param $variables
+   *   Array of variables to replace in the message on display or NULL if
+   *   message is already translated or not possible to translate.
+   * @param $severity
+   *   The severity of the message; one of the following values:
+   *     - TRIPAL_CRITICAL: Critical conditions.
+   *     - TRIPAL_ERROR: Error conditions.
+   *     - TRIPAL_WARNING: Warning conditions.
+   *     - TRIPAL_NOTICE: Normal but significant conditions.
+   *     - TRIPAL_INFO: (default) Informational messages.
+   *     - TRIPAL_DEBUG: Debug-level messages.
+   */
+  public function logMessage($message, $variables = [], $severity = TRIPAL_INFO) {
+    // Generate a translated message.
+    $tmessage = t($message, $variables);
+
+    // If we have a job then pass along the messaging to the job.
+    if ($this->job) {
+      // $this->job->logMessage($message, $variables, $severity);
+      $this->job->log($message);
+    }
+    // If we don't have a job then just use the drpual_set_message.
+    else {
+      // Report this message to watchdog or set a message.
+      if ($severity == TRIPAL_CRITICAL or $severity == TRIPAL_ERROR) {
+        drupal_set_message($tmessage, 'error');

```

Comment:
Drupal_set_message is deprecated in 9.0.x. See https://www.drupal.org/node/2774931 for how to change.